pkg/cmd/label/clone.go

```diff
@@ -105,32 +107,67 @@ func cloneRun(opts *cloneOptions) error {
 	return nil
 }
 
-func cloneLabels(client *http.Client, source, destination ghrepo.Interface, force bool) (successCount, totalCount int, err error) {
+func cloneLabels(client *http.Client, destination ghrepo.Interface, opts *cloneOptions) (successCount, totalCount int, err error) {
 	successCount = 0
-	labels, totalCount, err := listLabels(client, source, listQueryOptions{Limit: -1})
+	labels, totalCount, err := listLabels(client, opts.SourceRepo, listQueryOptions{Limit: -1})
 	if err != nil {
 		return
 	}
 
-	createOpts := createOptions{
-		Force: force,
+	workers := 10
+	ctx, abortCreate := context.WithCancel(context.Background())
+	defer abortCreate()
+	toCreate := make(chan createOptions)
+	created := make(chan interface{})
+
+	wg := new(errgroup.Group)
+	for i := 0; i < workers; i++ {
+		wg.Go(func() error {
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case l, ok := <-toCreate:
+					if !ok {
+						return nil
+					}
+					createErr := createLabel(client, destination, &l)
+					if createErr != nil {
+						if !errors.Is(createErr, errLabelAlreadyExists) {
+							abortCreate()
+							err := createErr
+							return err
+						}
+					} else {
+						created <- nil
+					}
+				}
+			}
+		})
 	}
 
-	for _, label := range labels {
-		createOpts.Name = label.Name
-		createOpts.Description = label.Description
-		createOpts.Color = label.Color
-
-		createErr := createLabel(client, destination, &createOpts)
-		if createErr != nil {
-			if !errors.Is(createErr, errLabelAlreadyExists) {
-				err = createErr
-				return
-			}
-		} else {
+	doneCh := make(chan struct{})
+	go func() {
+		for range created {
 			successCount++
 		}
+		close(doneCh)
+	}()
```

Comment:
@samcoe this is the same pattern you added to `status` but I think there's a potential deadlock issue here. In trying to figure out why the mac test failed (it's still not clear since that's the only API that would return an error in the `errgroup.Group`), I found this would occasionally deadlock in an error case. I think we also need to select `ctx.Done()` and/or check if the channel was closed (might actually be redundant). I'm pushing another commit that shows this. See what you think, and I'd be happy to add another commit to fix `status` or open another PR.

Reply:
I appreciate the Go axiom "share by communicating", but for just incrementing a counter perhaps this is a bit convoluted? I still think the new pattern might be good to use for `status` but what about just using a mutex here or `atomic.AddUint32()` from `sync/atomic`? The `errgroup` support would still be valuable - just thinking about eliminating the separate goroutine to handling incrementing the counter.